### DISCUSSION TITLE
Spawning the language server with cmd: lua-language-server failed. The language server is not either installed, missing the path or not executable.

### DISCUSSION THREAD

**Discussion by LeoGitHuber** [williamboman/nvim-lsp-installer#509] [Q&A] [answered]
I installed sumneko_lua by nvim-lsp-installer. But it doesn't work properly. The configuration of nvim-lsp-installer is default. This problem happened in Windows and EndeavourOS. Do I need to add path of language server manually?

**Comment by williamboman** [comment, marked as answer]
You don't have to manage paths manually with nvim-lsp-installer. When setting up servers via nvim-lsp-installer, the necessary PATH will be managed for you. How are you setting up servers? See https://github.com/williamboman/nvim-lsp-installer#setup for a minimum setup example

**Reply by yifan0414** [reply]
I'm having the same problem. While I install the sumneko_lua by using the nvim-lsp-installer.
![Xnip2022-03-01_00-12-13](https://user-images.githubusercontent.com/45898625/156017929-5bd35214-a0e5-4f06-9943-daa204fafe87.jpg)
Then the terminal show the sumneko_lua install successfully
![Xnip2022-03-01_00-12-57](https://user-images.githubusercontent.com/45898625/156018101-eddd11ea-e11b-4e3b-9481-aa57596789b2.jpg)
but when I nvim the .lua file, it still notice error.
![Xnip2022-03-01_00-16-28](https://user-images.githubusercontent.com/45898625/156018265-5f15c8bd-e98e-4289-92b6-760774d7a461.jpg)
how can i do, please

**Reply by yifan0414** [reply]
and the lsp.lua 
```lua
local sumneko_binary_path = vim.fn.exepath("lua-language-server")
if vim.g.is_mac or vim.g.is_linux and sumneko_binary_path ~= "" then
  local sumneko_root_path = vim.fn.fnamemodify(sumneko_binary_path, ":h:h:h")

  local runtime_path = vim.split(package.path, ";")
  table.insert(runtime_path, "lua/?.lua")
  table.insert(runtime_path, "lua/?/init.lua")

  lspconfig.sumneko_lua.setup({
    on_attach = custom_attach,
    cmd = { sumneko_binary_path, "-E", sumneko_root_path .. "/main.lua" },
    settings = {
      Lua = {
        runtime = {
          -- Tell the language server which version of Lua you're using (most likely LuaJIT in the case of Neovim)
          version = "LuaJIT",
          -- Setup your lua path
          path = runtime_path,
        },
        diagnostics = {
          -- Get the language server to recognize the `vim` global
          globals = { "vim" },
        },
        workspace = {
          -- Make the server aware of Neovim runtime files
          library = api.nvim_get_runtime_file("", true),
        },
        -- Do not send telemetry data containing a randomized but unique identifier
        telemetry = {
          enable = false,
        },
      },
    },
    capabilities = capabilities,
  })
end
```

**Reply by cserb** [reply]
same here

**Reply by williamboman** [reply]
It seems like you're setting up sumneko_lua manually and provide the `cmd` yourself. Have you read https://github.com/williamboman/nvim-lsp-installer#setup yet?

**Reply by williamhCode** [reply]
None of my installs work too, and everytime I go into a file (python, lua, etc.) I have a server installed for, I get the same error.

**Reply by williamhCode** [reply]
I required my nvim-lsp-installer config before the lspconfig config and it worked

**Reply by tea0112** [reply]
> I required my nvim-lsp-installer config before the lspconfig config and it worked

Today I've had the same issue as you. Just place its setup config before any language server, and it worked.

**Reply by tagrudev** [reply]
That solved it for me as well @williamhCode thank you.

**Reply by Pheon-Dev** [reply]
Use [Mason](https://github.com/williamboman/mason.nvim/) instead of [nvim-lsp-installer](https://github.com/williamboman/nvim-lsp-installer) and make sure you use ``PATH =  "prepend" -- instead of "skip"`` in your config as illustrated below:
```
require("mason").setup({
    PATH = "prepend", -- "skip" seems to cause the spawning error
})
```
